### PR TITLE
Add cancel method for PINRemoteImageManager

### PR DESCRIPTION
**Pull request by zhongwuzw**
Add cancel all tasks for `PINRemoteImageManager`, it's the convenience method for user to call all tasks one shot, don't need to maintain uuids by themself.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/b18ad9e38de71b6b54dc57c2d9f2a0ac/raw/b0bda9486c0951efb90d1a4ef74c1050c9037337/log.txt)

**Comment by bolsinga** [comment]
Please add tests.

**Comment by garrettmoon** [comment]
Hmm, I'm not sure this will work. If the UUIDs are stored in a weak hash table, PINRemoteImageManager will 'lose' references to requests that the caller doesn't strongly retain the UUID to…

**Comment by zhongwuzw** [comment]
@garrettmoon 🤔 Emm, `PINRemoteImageTask` would retain `UUID` by `_callbackBlocks`.

**Comment by zhongwuzw** [comment]
@garrettmoon And I may have a thought to optimize `_locked_taskForUUID:key:` of `PINRemoteManager` by add a weak to weak `NSMapTable`(uuid to task) and exposure `key` of `PINRemoteImageTask`, what do you think? 🤔

**Comment by garrettmoon** [comment]
> @garrettmoon 🤔 Emm, `PINRemoteImageTask` would retain `UUID` by `_callbackBlocks`.

You're right! Would you mind adding a comment to the code saying as much?

**Comment by garrettmoon** [comment]
> @garrettmoon And I may have a thought to optimize `_locked_taskForUUID:key:` of `PINRemoteManager` by add a weak to weak `NSMapTable`(uuid to task) and exposure `key` of `PINRemoteImageTask`, what do you think? 🤔

Seems like a good optimization because it will also make the code clearer I think!

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by zhongwuzw** [comment]
> > @garrettmoon And I may have a thought to optimize `_locked_taskForUUID:key:` of `PINRemoteManager` by add a weak to weak `NSMapTable`(uuid to task) and exposure `key` of `PINRemoteImageTask`, what do you think? 🤔
> 
> Seems like a good optimization because it will also make the code clearer I think!

👌 I'll make a PR later~

File: Source/Classes/PINRemoteImageManager.m [hunk 1]
```diff
@@ -695,6 +697,8 @@ - (NSUUID *)downloadImageWithURL:(NSURL *)url
             }
             [task addCallbacksWithCompletionBlock:completion progressImageBlock:progressImage progressDownloadBlock:progressDownload withUUID:UUID];
             [self.tasks setObject:task forKey:key];
+            // Relax :), task retain the UUID for us, it's ok to have a weak reference to UUID here.
```
Comment (garrettmoon) [inline Source/Classes/PINRemoteImageManager.m]: Great comment :)

**Pull request merged**